drivers/serial/uart_mcux_lpuart.c

```diff
@@ -5,6 +5,8 @@
  */
 
 #include <errno.h>
+#include <soc.h>
+#include <fsl_lpuart.h>
```

Comment:
these were copied, not moved, so they're still below.

Reply:
Thanks. Trying once more :)

Reply:
Let's see now

> 
> 
> @ioannisg The reason you need to cherry-pick [pabigot@ea951b4](https://github.com/pabigot/zephyr/commit/ea951b46e91adaafecb78c3cf110185e8737516e) and remove your NXP change from the stm32 commit is that NXP defines ARRAY_SIZE if it doesn't have it already, while sys/utils.h defines it unconditionally. So soc.h has to go first, then the NXP header, then devices.h.
> 
> kw41z still fails because of this.

Right. The reason for my original change was that I wanted the driver structure definition to go before kernel header inclusions, due to the issue with the linker-def macro that had the same name as a variable in a driver struct (see my other comment)